sopel/coretasks.py

```diff
@@ -35,6 +36,46 @@
 who_reqs = {}  # Keeps track of reqs coming from this module, rather than others
 
 
+def setup(bot):
+    bot.memory['join_events_queue'] = collections.deque()
+
+    # Manage JOIN flood protection
+    if bot.settings.core.throttle_join:
+        wait_interval = max(bot.settings.core.throttle_wait, 1)
+
+        @module.interval(wait_interval)
+        def processing_job(bot):
+            _join_event_processing(bot)
+
+        loader.clean_callable(processing_job, bot.settings)
+        bot.scheduler.add_job(jobs.Job(wait_interval, processing_job))
+
+
+def shutdown(bot):
+    try:
+        bot.memory['join_events_queue'].clear()
+    except KeyError:
+        pass
+
+
+def _join_event_processing(bot):
+    """Process a batch of JOIN event from the ``join_events_queue`` queue.
+
+    Every time this function is executed, it processes at most
+    ``throttle_join`` JOIN events. For each JOIN, it sends a WHO request to
+    know more about the channel. This will prevent an excess of flood when
+    there are too many channels to join at once.
+    """
+    batch_size = max(bot.settings.core.throttle_join, 1)
+    for _ in range(batch_size):
+        try:
+            channel = bot.memory['join_events_queue'].popleft()
+        except IndexError:
+            break
+        LOGGER.debug('Sending WHO after JOIN channel: %s', channel)
```

Comment:
I don't like this new wording either, even though it's my own. Sorry! 😹

```suggestion
        LOGGER.debug('Sending WHO after channel JOIN: %s', channel)
```